dll/win32/shell32/dialogs/drvdefext.cpp

```diff
@@ -745,37 +769,75 @@ CDrvDefExt::GetCommandString(UINT_PTR idCmd, UINT uType, UINT *pwReserved, LPSTR
     return E_NOTIMPL;
 }
 
-HRESULT WINAPI
-CDrvDefExt::AddPages(LPFNADDPROPSHEETPAGE pfnAddPage, LPARAM lParam)
+HRESULT
+CDrvDefExt::AddMainPage(LPFNADDPROPSHEETPAGE pfnAddPage, LPARAM lParam)
 {
-    HPROPSHEETPAGE hPage;
+    WCHAR szTitle[MAX_PATH], *pszTitle = NULL;
+    if (m_Multiple)
+    {
+        CComHeapPtr<ITEMIDLIST_ABSOLUTE> pidl(SHSimpleIDListFromPathW(m_wszDrive));
+        if (SUCCEEDED(SHGetNameAndFlagsW(pidl, SHGDN_INFOLDER, szTitle, _countof(szTitle), NULL)))
+            pszTitle = szTitle;
+    }
 
+    HPROPSHEETPAGE hPage;
     hPage = SH_CreatePropertySheetPageEx(IDD_DRIVE_PROPERTIES, GeneralPageProc, (LPARAM)this,
-                                         NULL, &PropSheetPageLifetimeCallback<CDrvDefExt>);
+                                         pszTitle, &PropSheetPageLifetimeCallback<CDrvDefExt>);
     HRESULT hr = AddPropSheetPage(hPage, pfnAddPage, lParam);
     if (FAILED_UNEXPECTEDLY(hr))
         return hr;
     else
         AddRef(); // For PropSheetPageLifetimeCallback
+    return hr;
+}
+
+HRESULT WINAPI
+CDrvDefExt::AddPages(LPFNADDPROPSHEETPAGE pfnAddPage, LPARAM lParam)
+{
+    HRESULT hr = AddMainPage(pfnAddPage, lParam);
+    if (FAILED_UNEXPECTEDLY(hr))
+        return hr;
 
-    if (GetDriveTypeW(m_wszDrive) == DRIVE_FIXED)
+    if (m_Multiple)
     {
-        hPage = SH_CreatePropertySheetPage(IDD_DRIVE_TOOLS,
-                                           ExtraPageProc,
-                                           (LPARAM)this,
-                                           NULL);
-        if (hPage)
-            pfnAddPage(hPage, lParam);
+        CDrop drop(m_Multiple);
+        UINT count = SUCCEEDED(drop.hr) ? drop.GetCount() : 0;
+        for (UINT i = 0; ++i < count;) // Skipping the first drive since it already has a page
```

Comment:
Why not then:
```suggestion
        for (UINT i = 1; i < count; ++i) // Skipping the first drive since it already has a page
```
(to have a more customary-looking for-loop)

Reply:
Why would I want it to look like a customary loop when it's not?